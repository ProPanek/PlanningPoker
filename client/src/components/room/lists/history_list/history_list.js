import React, {Component} from "react";
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import styled from "styled-components";
import ExpansionPanel from "@material-ui/core/ExpansionPanel/ExpansionPanel";
import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails";
import Divider from "@material-ui/core/Divider/Divider";
import Card from "@material-ui/core/Card";

const JiraDiv = styled.div`
  height:50%;
  overflow-y:auto;
`;

class HistoryList extends Component {
  render() {
    return (
        <JiraDiv>
          {this.props.store.room.cardHistory.length > 0 && <Typography>Jira Cards History</Typography>}
          {this.props.store.room.cardHistory.length > 0 && this.props.store.room.cardHistory.map((data, index) => {
            return (
              <ExpansionPanel key={index}>
                <ExpansionPanelSummary expandIcon={<ExpandMoreIcon/>}>
                  {`deal ${index}`}
                </ExpansionPanelSummary>
                <ExpansionPanelDetails style={{display: "flex", flexWrap:"wrap"}}>
                  {data.map((data) => (
                    <Card style={{display: "flex", flexDirection: "column", textAlign: "center", margin: "5px"}}>
                      <div style={{color: "black"}}>
                        {data.userName}
                      </div>
                      <Divider/>
                      <div style={{textAlign: "center", color: "black"}}>
                        {data.cardValue}
                      </div>
                    </Card>)
                  )}
                </ExpansionPanelDetails>
              </ExpansionPanel>
            );
          })}
        </JiraDiv>
    )
  }
}

export {HistoryList}
export default inject("store")(withRouter(observer(HistoryList)));

