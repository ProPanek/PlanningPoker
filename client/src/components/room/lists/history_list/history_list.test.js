import React from "react"
import {HistoryList} from "./history_list"
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from "sinon";

configure({adapter: new Adapter()});

describe("HistoryList", () => {
  it("renders without errors", () => {
    const props = {
      store: {
        room: {cardsAreTheSame: false, cardResults: [], cardHistory: []},
        user: {admin: true, users: []},
        jira: {activeBoardFetching: false, activeBoard: {issues: []}}
      }
    }

    const component = shallow(<HistoryList {...props} />)
    expect(component).toMatchSnapshot()
  })

  it("selects room", () => {
    const props = {
      store: {
        selectBoard: sinon.spy(),
        room: {cardsAreTheSame: false, cardResults: [], cardHistory: []},
        jira: {boardId:""}
      },
    }

    const component = shallow(<HistoryList {...props}/>)
    const instance = component.instance()
    const e = {target:{value:0}}
    instance.selectBoard(e)
    expect(props.store.selectBoard.called).toBe(true)
  });
})
