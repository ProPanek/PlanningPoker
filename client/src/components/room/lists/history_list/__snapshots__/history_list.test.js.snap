// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList renders without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HistoryList
    store={
      Object {
        "jira": Object {
          "activeBoard": Object {
            "issues": Array [],
          },
          "activeBoardFetching": false,
        },
        "room": Object {
          "cardHistory": Array [],
          "cardResults": Array [],
          "cardsAreTheSame": false,
        },
        "user": Object {
          "admin": true,
          "users": Array [],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  height:50%;
  overflow-y:auto;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  height:50%;
  overflow-y:auto;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
